packages/knip/src/plugins/hardhat/index.ts

```diff
@@ -0,0 +1,25 @@
+import { toDependency } from 'src/util/input.js';
+import type { IsPluginEnabled, Plugin, Resolve } from '../../types/config.js';
+import { hasDependency } from '../../util/plugin.js';
+
+// https://hardhat.org/docs
+
+const title = 'Hardhat';
+
+const enablers = ['hardhat'];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const entry: string[] = ['hardhat.config.{js,cjs,mjs,ts}'];
+
+const resolve: Resolve = async () => {
+  return [toDependency('hardhat')];
+};
```

Comment:
> The `hardhat` package may or may not be imported.

I understand the reasoning here, but if it's indeed not imported anywhere, then perhaps it's used in some other way? E.g. in `package.json#scripts` or somewhere else?

Reply:
Some people add scripts, but there's also a `hardhat-shorthand` package which is installed globally and adds an `hh` shortcut to replace `npx hardhat`, `yarn run hardhat`, etc.  It parses the Hardhat config and supports tab completion for builtin and user-defined commands, making it categorically different from scripts.

Reply:
Thanks, bit of a bummer but let's roll with it :)